src/vcpkg/base/git.cpp

```diff
@@ -199,4 +194,59 @@ namespace vcpkg
                            Tools::GIT)
             .map([](std::string&& output) { return "true" == Strings::trim(std::move(output)); });
     }
+
+    ExpectedL<std::string> git_head_sha(const GitConfig& config, StringView refname)
+    {
+        const auto cmd_line = git_cmd_builder(config).string_arg("rev-parse").string_arg(refname);
+        return flatten_out(cmd_execute_and_capture_output(cmd_line), Tools::GIT)
+            .map([](std::string&& output) {
+                Strings::inplace_trim(output);
+                return std::move(output);
+            })
+            .map_error([&](LocalizedString&& err) {
+                return msg::format(msgGitCommandFailed, msg::command_line = cmd_line.command_line())
+                    .append_raw('\n')
+                    .append(std::move(err));
+            });
+    }
+
+    ExpectedL<Unit> git_fetch(const Filesystem& fs, const GitConfig& config, StringView repo, StringView treeish)
+    {
+        const auto& work_tree = config.git_work_tree;
+        fs.create_directories(work_tree, VCPKG_LINE_INFO);
+
+        auto lock_file = work_tree / ".vcpkg-lock";
+
+        auto guard = fs.take_exclusive_file_lock(lock_file, stderr_sink, IgnoreErrors{});
+
+        auto init_registries_git_dir = git_cmd_builder(config).string_arg("init");
+        auto maybe_init_output = flatten(cmd_execute_and_capture_output(init_registries_git_dir), Tools::GIT);
+        if (!maybe_init_output)
```

Comment:
This function does not appear to be well described either if it's potentially doing `init`.